test/rpc_test.py

```diff
@@ -1319,25 +1319,35 @@ def test_debug_info(self):
         expected.update(autograd_info)
         self.assertEqual(expected, info)
 
-    @requires_process_group_agent("Testing exceptions thrown by ProcessGroupAgent::enqueueSend.")
+    @dist_init(clean_shutdown=False)
     def test_sender_exceptions(self):
-        rpc.init_rpc(
-            name="worker%d" % self.rank,
-            backend=rpc.backend_registry.BackendType[
-                dist_utils.TEST_CONFIG.rpc_backend_name
-            ],
-            rank=self.rank,
-            world_size=self.world_size,
-            rpc_backend_options=self.rpc_backend_options,
-        )
+        # This barrier is needed to ensure that some workers do not exit before
+        # others have been brought up, for non ProcessGroupAgent backends.
+        if not dist.is_initialized():
+            dist.init_process_group(
+                backend="gloo",
+                init_method=self.init_method,
+                rank=self.rank,
+                world_size=self.world_size,
+            )
+        dist.barrier()
 
         if self.rank == 0:
-            time.sleep(3)
-            # allow other workers to exit without joining
-            fut = rpc.rpc_async("worker1", torch.add, args=(torch.ones(1), 3))
-            with self.assertRaisesRegex(RuntimeError, "Encountered exception in ProcessGroupAgent::enqueueSend"):
+            dst_rank = (self.rank + 1) % self.world_size
+            self_worker = "worker{}".format(self.rank)
+            dst_worker = "worker{}".format(dst_rank)
+            # allow destination worker to exit without joining
+            wait_until_node_failure(dst_rank)
+            fut = rpc.rpc_async(dst_worker, torch.add, args=(torch.ones(1), 3))
+            error_str = (
+                "Encountered exception in ProcessGroupAgent::enqueueSend"
+                if self.rpc_backend == rpc.backend_registry.BackendType.PROCESS_GROUP
+                else "{}: Error in response from {}".format(self_worker, dst_worker)
```

Comment:
Can you run this test multiple times and verify you always get this error? I saw that the thrift agent could report two types of errors. See https://github.com/pytorch/pytorch/pull/31508/files#diff-9c5747e2e5b79a428297f80897df2ce3R1319

Reply:
Added the other error regex as per in that diff.